client/project_policy.go

```diff
@@ -1,68 +1,35 @@
+	DisableDestroyEnvironments  bool    `json:"disableDestroyEnvironments"`
+	SkipRedundantDeployments    bool    `json:"skipRedundantDeployments"`
+	UpdatedBy                   string  `json:"updatedBy"`
+	RunPullRequestPlanDefault   bool    `json:"runPullRequestPlanDefault"`
+	ContinuousDeploymentDefault bool    `json:"continuousDeploymentDefault"`
+	MaxTtl                      *string `json:"maxTtl,omitempty"`
+	DefaultTtl                  *string `json:"defaultTtl,omitempty"`
 }
 
 type PolicyUpdatePayload struct {
 	ProjectId                   string `json:"projectId"`
-	NumberOfEnvironments        int    `json:"numberOfEnvironments"`
-	NumberOfEnvironmentsTotal   int    `json:"numberOfEnvironmentsTotal"`
+	NumberOfEnvironments        *int   `json:"numberOfEnvironments,omitempty"`
+	NumberOfEnvironmentsTotal   *int   `json:"numberOfEnvironmentsTotal,omitempty"`
 	RequiresApprovalDefault     bool   `json:"requiresApprovalDefault"`
 	IncludeCostEstimation       bool   `json:"includeCostEstimation"`
 	SkipApplyWhenPlanIsEmpty    bool   `json:"skipApplyWhenPlanIsEmpty"`
 	DisableDestroyEnvironments  bool   `json:"disableDestroyEnvironments"`
 	SkipRedundantDeployments    bool   `json:"skipRedundantDeployments"`
 	RunPullRequestPlanDefault   bool   `json:"runPullRequestPlanDefault"`
 	ContinuousDeploymentDefault bool   `json:"continuousDeploymentDefault"`
-}
-
-func (p PolicyUpdatePayload) MarshalJSON() ([]byte, error) {
-	type serial struct {
-		ProjectId                   string `json:"projectId"`
-		NumberOfEnvironments        *int   `json:"numberOfEnvironments"`
-		NumberOfEnvironmentsTotal   *int   `json:"numberOfEnvironmentsTotal"`
-		RequiresApprovalDefault     bool   `json:"requiresApprovalDefault"`
-		IncludeCostEstimation       bool   `json:"includeCostEstimation"`
-		SkipApplyWhenPlanIsEmpty    bool   `json:"skipApplyWhenPlanIsEmpty"`
-		DisableDestroyEnvironments  bool   `json:"disableDestroyEnvironments"`
-		SkipRedundantDeployments    bool   `json:"skipRedundantDeployments"`
-		RunPullRequestPlanDefault   bool   `json:"runPullRequestPlanDefault"`
-		ContinuousDeploymentDefault bool   `json:"continuousDeploymentDefault"`
-	}
-
-	s := serial{
-		ProjectId:                   p.ProjectId,
-		RequiresApprovalDefault:     p.RequiresApprovalDefault,
-		IncludeCostEstimation:       p.IncludeCostEstimation,
-		SkipApplyWhenPlanIsEmpty:    p.SkipApplyWhenPlanIsEmpty,
-		DisableDestroyEnvironments:  p.DisableDestroyEnvironments,
-		SkipRedundantDeployments:    p.SkipRedundantDeployments,
-		RunPullRequestPlanDefault:   p.RunPullRequestPlanDefault,
-		ContinuousDeploymentDefault: p.ContinuousDeploymentDefault,
-	}
-
-	if p.NumberOfEnvironments != 0 {
-		s.NumberOfEnvironments = &p.NumberOfEnvironments
-	}
-	if p.NumberOfEnvironmentsTotal != 0 {
-		s.NumberOfEnvironmentsTotal = &p.NumberOfEnvironmentsTotal
-	}
-
-	return json.Marshal(s)
+	MaxTtl                      string `json:"maxTtl"`
```

Comment:
For the update case these are not pointers - because the schema defaults to "inherit" value will always be available.